### ISSUE TITLE
Feature request: Make the site generation part a stand-alone tool

### ISSUE THREAD

**Issue by ericsagnes** [NixOS/nixos-weekly#23]
A nix based static site generation is very promising and could be used in many other projects. (I would love to replace some hakyll sites I have with this)
#22 is an effort to make the site generation more generic. If this can be made generic enough this project base code could be turned in a stand-alone static site generator.

It would need a name, and maybe an executable to generate a base project like other static site generators do.

**Comment by ericsagnes** [comment]
I am thinking to start documenting the site generation, any preference about the format?
Shall we go for docbook like other nix projects or just simple markdown?

My personal preference would go to docbook as call outs can really be handy, but I agree its verbosity can be problematic.

**Comment by zimbatm** [comment]
Are you going to use the site generator to generate the documentation? :)

**Comment by zimbatm** [comment]
Asciidoc(tor) was designed to replace docbook and doesn't have it's verbosity issues. It can do things like includes and TOC generation that become useful in bigger document structures.

**Comment by Mic92** [comment]
Also interest in this context: the linux documentation project which switched to asciidoc: https://lwn.net/Articles/692704/ -

**Comment by ericsagnes** [comment]
I made a [prototype](https://github.com/ericsagnes/styx) (very inspired by jekyll).
Name was randomly chosed from a pluto moon name like some of Nix related projects.
Feedback is welcomed :)

**Comment by ericsagnes** [comment]
> Asciidoc(tor) was designed to replace docbook and doesn't have it's verbosity issues. It can do things like includes and TOC generation that become useful in bigger document structures.

Thanks! That was a very helpful advice, asciidoc is very pleasant to write and has all the functionalities I needed.

**Comment by zimbatm** [comment]
I think it's really cool what you did. It looks easy to use and to read.

The next level of power would be to wrap it so that it's not necessary to ship styx. For example if it's made part of nixpkgs. Then just need a top-level default.nix and the content of the site are needed. The default.nix would look like:

``` nix
with import <nixpkgs> {};
stdenv.mkStyxSite {
  name = "my-site-name";
  conf = import ./conf.nix; # or keep it inline
  src = ./.;
  # ...
}
```

**Comment by ericsagnes** [comment]
That would indeed move easiness to use one step further.

One possible problem for this is that actually the site generation process is not deterministic as it take the generation time timestamp as an input to set the `<updated>` value of the feed. I wonder if there is a clean way to achieve that.

By the way, I opened an issue for this suggestion https://github.com/ericsagnes/styx/issues/2.

**Comment by ericsagnes** [comment]
The stand-alone tool (Styx) is close to a first [release](https://github.com/ericsagnes/styx/issues/5).

I would like to set a license on Styx, but I am not sure what are the implications as it can probably be considered as a fork of this project as it share some of its code.

I am thinking to make it MIT, but I have no idea if it is fine.

This site says CC-BY-SA-4.0, but my understanding is that it is for the content and this does not cover the source code.

In short, what is the license of nixos-weekly source code, and is it fine to release Styx with a MIT license?

**Comment by zimbatm** [comment]
Sure, choose whatever license you want. nixpkgs itself is under the MIT.

**Comment by ericsagnes** [comment]
Thanks! made it MIT.

**Comment by ericsagnes** [comment]
Closed via https://github.com/styx-static/styx/commit/3ad814d0eadfc4a96e7d92a3e713b041564a3695

And it has a [site](https://styx-static.github.io/styx-site/) made with Styx :)

**Comment by zimbatm** [comment]
Cool.

The design is a bit broken on NixOS for me (same in FF): ![2016-10-07-095333_1280x720_scrot](https://cloud.githubusercontent.com/assets/3248/19184211/fae8d09a-8c73-11e6-88f2-d0c70372e8c6.png)

**Comment by ericsagnes** [comment]
Thanks, it was missing some cross domain css thing (?), added bootstrap javascript too so it should display fine on mobile devices too.
